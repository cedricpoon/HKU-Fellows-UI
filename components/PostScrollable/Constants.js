export const FADE_IN_DURATION = 250;
