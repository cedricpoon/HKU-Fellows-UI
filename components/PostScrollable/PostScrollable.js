import React, { Component } from 'react';
import PropTypes from "prop-types";
import * as Animatable from 'react-native-animatable';
import { FlatList as RNFlatList } from 'react-native';

import Post from './PostPreview/PostPreview';

const FlatList = Animatable.createAnimatableComponent(RNFlatList);

class PostScrollable extends Component {

  _renderPost({item}) {
    return (
      <Post
        id={item.id}
        {...item}
      />
    );
  }

  render() {
    const { posts, ...restProps } = this.props

    if (this.props.posts.length > 0) {
      return (
        <FlatList
          data={posts}
          keyExtractor={item => item.id}
          renderItem={this._renderPost}
          animation='fadeIn'
          duration={FADE_IN_DURATION}
          {...restProps}
        />
      );
    } else {
      return null;
    }
  }
}

PostScrollable.propTypes = {
  posts: PropTypes.array.isRequired
};

export const FADE_IN_DURATION = 250;

export default PostScrollable;
