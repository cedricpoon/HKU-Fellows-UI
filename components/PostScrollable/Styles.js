import nbColor from "hkufui/native-base-theme/variables/commonColor";
import { logo } from "hkufui/theme/palette";
import { post, gridBase } from "hkufui/theme/grid"
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  noPost: {
    alignItems:'center',
    justifyContent:'center'
  },
  noMarginLeft: {
    marginLeft: 0
  },
  badgeContainer: {
    width: post.seperatorPadding * 2,
  },
  moodle: {
    color: logo.blue
  },
  hashtags: {
    color: logo.green
  },
  note: {
    color: nbColor.listNoteColor
  },
  active: {
    color: logo.red
  },
  infoBar: {
    flexDirection: 'row'
  },
  topBadge: {
    color: logo.yellow,
    fontSize: post.smallSize,
    paddingLeft: gridBase * 1.5
  },
  right: {
    justifyContent: 'flex-end'
  },
  last: {
    paddingRight: post.seperatorPadding
  },
  mainText: {
    fontSize: post.titleSize,
    paddingBottom: gridBase
  },
  subText: {
    fontSize: post.miniSize,
    paddingBottom: gridBase,
    paddingTop: gridBase,
    paddingRight: gridBase
  }
});
