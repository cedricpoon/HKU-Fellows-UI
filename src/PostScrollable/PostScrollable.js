import React, { Component } from 'react';
import PropTypes from "prop-types";
import { List, ListItem, Body, Text, Icon } from 'native-base';

import { localize } from '../../locale';
import Post from './PostPreview/PostPreview';
import styles from './Styles';

const locale = localize({ language: 'en', country: 'hk' });

class PostScrollable extends Component {

  renderPosts(posts) {
    return posts.map((post, i) => (
      <Post
        key={i}
        {...post}
      />
    ));
  }

  renderEmpty() {
    return (
      <ListItem itemDivider>
        <Body style={styles.noPost}>
          <Text>{locale['post.noPostTitle']}</Text>
          <Text note>
            {locale['post.noPostContent']}
            <Icon name="md-paper" style={[styles.mainText, styles.note]} />
          </Text>
        </Body>
      </ListItem>
    );
  }

  render() {
    return (
      <List>
        {this.props.posts.length > 0 ?
          this.renderPosts(this.props.posts)
        :
          this.renderEmpty()
        }
      </List>
    );
  }
}

PostScrollable.propTypes = {
  posts: PropTypes.array.isRequired
};

export default PostScrollable;
