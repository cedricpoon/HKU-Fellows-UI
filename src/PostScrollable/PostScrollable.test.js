import React from 'react';
import { shallow } from 'enzyme';
import _ from 'enzyme-to-json';
import PostScrollable from './PostScrollable';

describe('Testing PostScrollable Component', () => {
  it('renders as expected with posts', () => {
    const wrapper = shallow(
      <PostScrollable posts={[]} />
    );
    expect(_(wrapper)).toMatchSnapshot();
  });
});
