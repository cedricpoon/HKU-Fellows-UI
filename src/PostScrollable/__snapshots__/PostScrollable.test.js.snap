// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostScrollable Component renders as expected with posts 1`] = `
<Styled(List)>
  <Styled(ListItem)
    itemDivider={true}
  >
    <Styled(Body)
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Styled(Text)>
        No Posts Yet
      </Styled(Text)>
      <Styled(Text)
        note={true}
      >
        Feel free to raise the First Post  
        <Styled(Icon)
          name="md-paper"
          style={
            Array [
              Object {
                "fontSize": 20,
                "paddingBottom": 5,
              },
              Object {
                "color": "#808080",
              },
            ]
          }
        />
      </Styled(Text)>
    </Styled(Body)>
  </Styled(ListItem)>
</Styled(List)>
`;
